### PR TITLE
get-articles

### PR DESCRIPTION
**Pull request by LW95x**
Please see comments in tests - looking for a little more clarity on what else I can test for. Thanks.

### CODE REVIEW

**Comment by LW95x** [comment]
To add - just realised that comments is a table in itself, I've added a count in my query for this using a join, but how can I validate that the comment_count is correct in my test? Do I need to create a query within my test of the comments table and compare these values?

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -94,3 +94,45 @@ describe("GET /api/articles/:article_id", () => {
       });
   });
 });
+
+describe("GET /api/articles", () => {
+  test("responds with a 200 status code", () => {
```
Comment (ghost) [inline __tests__/integration.test.js]: This should be combined in your other tests. This is integration testing, not unit testing. Its a ok and expected to combine parts such as the status code and the data that is returned.

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -94,3 +94,45 @@ describe("GET /api/articles/:article_id", () => {
...
+          expect(typeof property.topic).toBe("string");
+          expect(typeof property.author).toBe("string");
+          expect(typeof property.created_at).toBe("string");
+          expect(typeof property.votes).toBe("number");
+          expect(typeof property.article_img_url).toBe("string");
+          expect(typeof property.comment_count).toBe("string");
+        });
+      });
+  });
+  test("200: should respond with an array of articles without a body property present on any of the objects", () => {
```
Comment (ghost) [inline __tests__/integration.test.js]: This is also ok to combine with the above test in your forEach. I would refactor the type of to be objectContaining and then an extra toHaveProperty on the body to be false. Much more concise as the test above doesn't prove the body isn't there, but could do.

File: __tests__/integration.test.js [hunk 3]
```diff
@@ -94,3 +94,45 @@ describe("GET /api/articles/:article_id", () => {
...
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        body.articles.forEach((property) => {
+          expect(property.body).toBeUndefined();
+        });
+      });
+  });
+  test.todo(
```
Comment (ghost) [inline __tests__/integration.test.js]: If you were unsure of what to test this should have been an NCHelp not a pull request for review. You should be using jest sorted to test that your objects are returned in the correct order. The format is a correct utc date format which can be read by that package without any data manipulation.

File: db/models/getArticles.model.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const db = require("../connection");
+
+exports.findArticles = () => {
+    return db.query(`SELECT article_id, title, topic, author, created_at, votes, article_img_url, COUNT(body) AS comment_count FROM articles
```
Comment (ghost) [inline db/models/getArticles.model.js]: You won't be counting a body as the comment count. The comment count is meant to reflect how many comments have the same article id as the article in your list. You'll need to make a join.

**Review by ghost — CHANGES_REQUESTED** [review]
If you were unsure of how to proceed this should have been an nchelp not a pull request. This ticket is incomplete with missing tests and an incomplete sql join. Take a look at the notes for joins and aggregate functions to count the comments. 

For this endpoint you should be testing that:
[ ] There are the correct amount of articles
[ ] Each article does not have a body key
[ ] Each article has a comment count representing the amount of comments
[ ] Articles are sorted in date order

This can all be done in one test. Revisit this and resubmit. If you need direction, nchelp first.

**Review by Hard-Luck — APPROVED** [review]
Small optional (but recommended) change then merge

**Pull request merged**